pkg/builder/s2i.go

```diff
@@ -209,6 +209,7 @@ func (t *s2iTask) Do(ctx context.Context) v1.BuildStatus {
 	err = t.waitForS2iBuildCompletion(ctx, t.c, &s2iBuild)
 	if err != nil {
 		if errors.Is(err, context.Canceled) || errors.Is(err, context.DeadlineExceeded) {
+			// nolint: contextcheck
```

Comment:
not sure if the builder context should be passed down

Reply:
No it should not be passed here as the build context has been either cancel or has exceeded its deadline, so the cancel is a new operator with its own context.